tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -73,6 +73,7 @@ spec:
               echo "Max retries exceeded."
               exit 1
             fi
+            unset DOCKER_CONFIG
```

Comment:
`From second component on, it would always use the docker config that was crafted for the previous component` how does this help? Shouldn't line 62 run for each component?

Reply:
The problematic `get-image-architectures` is first called before the first call of this function (`push_image`) - line 128. It uses `skopeo`. `skopeo` doesn't need any magic with `select-oci-auth`, it just works with the stock ~/.docker/config.json that k8s creates from the secrets. So that's how it works the first time.

But then we run `push_image` which breaks it for next component. The reason it breaks it is that it sets env var DOCKER_CONFIG to a custom file that contains an auth entry for the previous component's full pull spec. So there is no generic quay.io/redhat-user-workloads entry in that file. So to  make sure that the next skopeo call uses the stock ~/.docker/config.json config, we need to unset the env var (we could also set it to that location).

Reply:
I think it makes sense to unset it, because that `cosign copy` is the only one that uses it. `oras` also needs custom docker config files, but we supply those as cli argument, not via env var.

Reply:
Gotcha, can you add a comment then? That isn't obvious to me. Something like `reset DOCKER_CONFIG as only cosign needs the custom one but it breaks skopeo` or something along those lines

Reply:
OK, added a comment.